recipes/think-cell-library/all/conanfile.py

```diff
@@ -0,0 +1,73 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import copy, get
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class PackageConan(ConanFile):
+    name = "think-cell-library"
+    description = "This library consists of several core C++ utilities that we at think-cell Software have developed and consider to be useful."
+    license = "BSL-1.0 license"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/think-cell/think-cell-library"
+    topics = ("ranges",  "header-only")
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+
+    @property
+    def _min_cppstd(self):
+        return 20
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "13",
+            "clang": "7",
+            "gcc": "12",
+            "msvc": "191",
+            "Visual Studio": "15",
+        }
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("boost/1.84.0")
```

Comment:
```suggestion
        self.requires("boost/1.84.0", transitive_headers=True, transitive_libs=True)
```
Or does package_type=header_library exposes automatically?

Reply:
It does! https://docs.conan.io/2/reference/conanfile/methods/requirements.html#default-traits-for-each-kind-of-requires